### PR TITLE
Added ability to export committee event informations as ICal format

### PR DESCRIPTION
**Pull request by xavismeh**
Choices made:
- added JMSSerializerBundle
- new ical visitor created
- `AppBundle\Entity\CommitteeEvent` handler (serialize only)

And for the frontend ... well... that's frontend ¯\_(ツ)_/¯

### CODE REVIEW

File: src/AppBundle/Serializer/Visitor/IcalSerializationVisitor.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+     */
+    public function getResult()
+    {
+        if (!$root = $this->getRoot()) {
+            return;
+        }
+
+        $vCalendar = new VCalendar($root);
+
+        return $vCalendar->serialize();
```
Comment (hhamon) [inline src/AppBundle/Serializer/Visitor/IcalSerializationVisitor.php]: `return (new VCalendar($root))->serialize();`

File: app/Resources/views/events/show.html.twig [hunk 2]
```diff
@@ -70,6 +70,14 @@
                         onclick="App.share('email', '{{ committee_event_url|url_encode }}', 'En Marche ! à {{ committee_event.cityName }} le {{ committee_event.beginAt|date('d/m') }}: ')">
                         <i class="fa fa-envelope-o"></i>
                     </li>
+                    <li role="button" arial-label="{{ 'committee.event.show.share.download_ical'|trans }}"
+                        class=""
```
Comment (tgalopin) [inline app/Resources/views/events/show.html.twig]: This seems weird

File: tests/AppBundle/Committee/Serializer/CommitteeEventICalHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
```
Comment (tgalopin) [inline tests/AppBundle/Committee/Serializer/CommitteeEventICalHandlerTest.php]: You don't need to put this on top of each file, Symfony already transform type errors into exceptions.

File: tests/AppBundle/Committee/Serializer/CommitteeEventICalHandlerTest.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /**
+     * Since the model does not require an organizer, contact informations should not be available
+     */
+    public function testSerializeNoOrganizer()
+    {
+        $visitor = $this->createMock(IcalSerializationVisitor::class);
+        $committeeEvent = $this->createMock(CommitteeEvent::class);
+        $uuid = $this->createMock('Ramsey\Uuid\UuidInterface');
```
Comment (tgalopin) [inline tests/AppBundle/Committee/Serializer/CommitteeEventICalHandlerTest.php]: `UuidInterface::class`

File: tests/AppBundle/Committee/Serializer/CommitteeEventICalHandlerTest.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        $type = [];
+        $serializationContext = $this->createMock('JMS\Serializer\SerializationContext');
```
Comment (tgalopin) [inline tests/AppBundle/Committee/Serializer/CommitteeEventICalHandlerTest.php]: `SerializationContext::class`

File: tests/AppBundle/Serializer/Visitor/IcalSerializationVisitorTest.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+{
+    protected $propertyNamingStrategy;
+    /**
+     * @var IcalSerializationVisitor
+     */
+    protected $visitor;
+
+    protected function setUp()
+    {
+        $this->propertyNamingStrategy = $this->createMock('JMS\Serializer\Naming\PropertyNamingStrategyInterface');
```
Comment (tgalopin) [inline tests/AppBundle/Serializer/Visitor/IcalSerializationVisitorTest.php]: `PropertyNamingStrategyInterface:class`

**Pull request closed without merging**